indico/modules/bootstrap/controllers.py

```diff
@@ -44,14 +47,26 @@ class RHBootstrap(RH):
     def _process_GET(self):
         if User.query.filter_by(is_system=False).has_rows():
             return redirect(url_for_index())
+
+        try:
+            hostname = urlparse(request.url_root).hostname.strip('[]')
+            host_data = socket.getaddrinfo(hostname, 80, 0, 0, socket.IPPROTO_TCP)
+            for item in host_data:
+                local_instance = ipaddress.ip_address(unicode(item[4][0])).is_private
+                if local_instance:
+                    break
+        except socket.gaierror, ipaddress.AddressValueError:
+            local_instance = True
+
         return render_template('bootstrap/bootstrap.html',
                                form=BootstrapForm(),
                                timezone=config.DEFAULT_TIMEZONE,
                                languages=get_all_locales(),
                                operating_system=get_os(),
                                postgres_version=get_postgres_version(),
                                indico_version=indico.__version__,
-                               python_version=python_version())
+                               python_version=python_version(),
+                               warn_before_hub_registration=local_instance or config.DEBUG)
```

Comment:
`show_local_warning`?

and IMO it's cleaner to wrap the value in `()`

- `foo=x or y`
- `foo=(x or y)`

with the latter it's clearer that everything there belongs to foo